components/ws-manager/pkg/manager/manager.go

```diff
@@ -184,27 +186,59 @@ func (m *Manager) StartWorkspace(ctx context.Context, req *api.StartWorkspaceReq
 	}
 	span.LogKV("event", "created start workspace context")
 	clog.Debug("starting new workspace")
-	// we must create the workspace pod first to make sure we don't clean up the services or configmap we're about to create
-	// because they're "dangling".
+
+	// create a Pod object for the workspace
 	pod, err := m.createWorkspacePod(startContext)
 	if err != nil {
 		return nil, xerrors.Errorf("cannot create workspace pod: %w", err)
 	}
 	span.LogKV("event", "pod description created")
-	err = m.Clientset.Create(ctx, pod)
-	if err != nil {
-		m, _ := json.Marshal(pod)
-		safePod, _ := log.RedactJSON(m)
 
-		if k8serr.IsAlreadyExists(err) {
-			clog.WithError(err).WithField("req", req).WithField("pod", safePod).Warn("was unable to start workspace which already exists")
-			return nil, status.Error(codes.AlreadyExists, "workspace instance already exists")
+	// create the Pod in the cluster and wait until is scheduled
+	// https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.22.md#workloads-that-saturate-nodes-with-pods-may-see-pods-that-fail-due-to-node-admission
+	backoff := wait.Backoff{
+		Steps:    10,
+		Duration: 100 * time.Millisecond,
+		Factor:   5.0,
+		Jitter:   0.1,
+	}
+
+	var retryErr error
+	err = wait.ExponentialBackoff(backoff, func() (bool, error) {
+		err = m.Clientset.Create(ctx, pod)
+		if err != nil {
+			m, _ := json.Marshal(pod)
+			safePod, _ := log.RedactJSON(m)
+
+			if k8serr.IsAlreadyExists(err) {
+				clog.WithError(err).WithField("req", req).WithField("pod", safePod).Warn("was unable to start workspace which already exists")
+				return false, status.Error(codes.AlreadyExists, "workspace instance already exists")
+			}
+
+			clog.WithError(err).WithField("req", req).WithField("pod", safePod).Error("was unable to start workspace")
+			return false, err
 		}
 
-		clog.WithError(err).WithField("req", req).WithField("pod", safePod).Error("was unable to start workspace")
-		return nil, err
+		err = wait.PollWithContext(ctx, 100*time.Millisecond, 5*time.Second, podRunning(m.Clientset, pod.Namespace, pod.Name))
+		if err != nil {
+			m, _ := json.Marshal(pod)
+			safePod, _ := log.RedactJSON(m)
+			clog.WithError(err).WithField("req", req).WithField("pod", safePod).Error("was unable to reach ready state")
+			retryErr = err
+			return true, nil
+		}
+
+		return true, nil
+	})
+	if err == wait.ErrWaitTimeout {
+		err = retryErr
 	}
-	span.LogKV("event", "pod created")
+
+	if err != nil {
+		return nil, xerrors.Errorf("cannot create workspace pod: %w", err)
+	}
+
+	span.LogKV("event", "pod created and scheduled")
```

Comment:
Would it make sense to say something like `"pod started successfully"` as we are already checking the podRunning status in previous steps?